### PR TITLE
port to torch 2.4 / python 3.12 (at least basic mono training works)

### PR DESCRIPTION
**Pull request by timjr**
update the resnet encoder for pytorch 2.4

update to torch 2.4

np.int is decprecated

### CODE REVIEW

**Comment by timjr** [comment]
this one has a stray learning rate step...
